src/query/storages/parquet/src/parquet2/processors/deserialize_transform.rs

```diff
@@ -148,6 +149,39 @@ impl Parquet2DeserializeTransform {
         Ok(())
     }
 
+    fn process_small_groups(&mut self, part: &Parquet2SmallGroupPart) -> Result<Vec<DataBlock>> {
+        let mut blocks = Vec::new();
+        for (path, data) in part.groups.iter() {
+            blocks.extend(self.process_small_group(path.as_str(), data)?);
+        }
+        Ok(blocks)
+    }
+
+    fn process_small_group(&mut self, path: &str, groups: &Vec<usize>) -> Result<Vec<DataBlock>> {
+        let mut res = Vec::new();
+        let builder = Memory::default();
+        let op = Operator::new(builder)?.finish();
+        let blocking_op = op.blocking();
+        let parts = self.partition_pruner.prune_file_group(path, &op, groups)?;
```

Comment:
`prune_file_group ` will read metadata, which should be avoided.
we should carry the info to read the row group in the PART.